azure-sink-connector/src/main/java/io/aiven/kafka/connect/azure/AzureBlobSinkConfig.java

```diff
@@ -0,0 +1,307 @@
+                            throw new ConfigException(FILE_NAME_TIMESTAMP_TIMEZONE, value, e.getMessage());
+                        }
+                    }
+                }, ConfigDef.Importance.LOW,
+                "Specifies the timezone in which the dates and time for the timestamp variable will be treated. "
+                        + "Use standard shot and long names. Default is UTC",
+                GROUP_FILE, fileGroupCounter++, ConfigDef.Width.SHORT, FILE_NAME_TIMESTAMP_TIMEZONE);
+
+        configDef.define(FILE_NAME_TIMESTAMP_SOURCE, ConfigDef.Type.STRING, TimestampSource.Type.WALLCLOCK.name(),
+                new ConfigDef.Validator() {
+                    @Override
+                    public void ensureValid(final String name, final Object value) {
+                        try {
+                            TimestampSource.Type.of(value.toString());
+                        } catch (final Exception e) { // NOPMD broad exception catched
+                            throw new ConfigException(FILE_NAME_TIMESTAMP_SOURCE, value, e.getMessage());
+                        }
+                    }
+                }, ConfigDef.Importance.LOW, "Specifies the timestamp variable source. Default is wall-clock.",
+                GROUP_FILE, fileGroupCounter, ConfigDef.Width.SHORT, FILE_NAME_TIMESTAMP_SOURCE);
+    }
+
+    public AzureBlobSinkConfig(final Map<String, String> properties) {
+        super(configDef(), handleDeprecatedYyyyUppercase(properties));
+        validate();
+    }
+
+    static Map<String, String> handleDeprecatedYyyyUppercase(final Map<String, String> properties) {
+        if (properties.containsKey(FILE_NAME_TEMPLATE_CONFIG)) {
+            final var result = new HashMap<>(properties);
+
+            String template = properties.get(FILE_NAME_TEMPLATE_CONFIG);
+            final String originalTemplate = template;
+
+            final var unitYyyyPattern = Pattern.compile("\\{\\{\\s*timestamp\\s*:\\s*unit\\s*=\\s*YYYY\\s*}}");
+            template = unitYyyyPattern.matcher(template)
+                    .replaceAll(matchResult -> matchResult.group().replace("YYYY", "yyyy"));
+
+            if (!template.equals(originalTemplate)) {
+                LOG.warn(
+                        "{{timestamp:unit=YYYY}} is no longer supported, "
+                                + "please use {{timestamp:unit=yyyy}} instead. " + "It was automatically replaced: {}",
+                        template);
+            }
+
+            result.put(FILE_NAME_TEMPLATE_CONFIG, template);
+
+            return result;
+        } else {
+            return properties;
+        }
+    }
+
+    private void validate() {
+        final String connectionString = getString(AZURE_STORAGE_CONNECTION_STRING_CONFIG);
+
+        if (connectionString == null) {
+            throw new ConfigException(
+                    String.format("The configuration %s cannot be null.", AZURE_STORAGE_CONNECTION_STRING_CONFIG));
+        }
+    }
```

Comment:
When defining the configuration, you can provide `ConfigDef.NO_DEFAULT_VALUE` rather than `null` to make the configuration required.